### ISSUE TITLE
Ldap configuration ok but users cannot login

### ISSUE THREAD

**Issue by pawel1615** [owncloud/core#24668]
### Steps to reproduce
1. Enter Ldap credentials
2. Test configuration looks ok, in Ldap server logs there is a Bind with entries=1
3. Ldap users cannot login - invalid password

### Expected behaviour
Ldap users should log in

### Actual behaviour
Local users works ok, ldap users - invalid password. in applications logs:
"PHP","message":"ldap_control_paged_result_response(): No server controls in result at \/var\/www\/html\/owncloud\/apps\/user_ldap\/lib\/ldap.php#70",

Could not get user object for DN uid=xxx,ou=people,o=domain,o=employees. Maybe the LDAP entry has no set display name attribute?
### Server configuration
**Operating system**:
Centos 6.7
**Web server:**
Apache 2.4
**Database:**
External Mysql
**PHP version:**
PHP 5.6.21
**ownCloud version:** (see ownCloud admin page)
Owncloud 8.2.3 stable
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
download tar.gz from offical owncloud website
**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
not available...
**List of activated apps:**
-encryption module
-external user support
-gallery
-ldap user and group backend
-pdf viewer
-API
-share filer
-versions
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
`Enabled:
  - encryption: 1.1.0
  - files: 1.2.1
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_versions: 1.1.0
  - gallery: 14.2.0
  - provisioning_api: 0.3.0
  - user_external: 0.4
  - user_ldap: 0.7.1
Disabled:
  - activity
  - external
  - files_external
  - files_texteditor
  - files_trashbin
  - files_videoviewer
  - firstrunwizard
  - notifications
  - templateeditor
  - updater
``

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'xxxxxxxxxxxx',
  'passwordsalt' => 'xxxxxxxxxxxxxxxxx',
  'secret' => 'xxxxxxxxxxxxxxxxx',
  'trusted_domains' => 
  array (
    0 => 'mydomain.com',
  ),
  'datadirectory' => '/var/www/html/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.2.3.2',
  'dbname' => 'xxxxx',
  'dbhost' => 'xxxxxxxl',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxxxxxx',
  'dbpassword' => 'xxxxxxxxxxxxx',
  'logtimezone' => 'UTC',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
);

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder
{
    "system": {
        "instanceid": "xxxxxxxx",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "mydomain.com"
        ],
        "datadirectory": "\/var\/www\/html\/owncloud\/data",
        "dbtype": "mysql",
        "version": "8.2.3.2",
        "dbname": "xxxxx",
        "dbhost": "xxxxxxxx",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "ldapIgnoreNamingRules": false

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
Oracle Ldap ( Oracle Internet Directory)

#### LDAP configuration (delete this part if not used)
**Server Tab**
host: myldapserver.com
DN: uid=proxy-user,ou=People,o=mydomain.com,o=employees
Password:xxx
Base DN:o=mydomain.com,o=employees
Manually enter ldap filter:checked
**Users**
(objectclass=person)
**Login Attributes**
(uid=%uid)
**groups**
blank
**Advanced**
Directory Settings
User Display Name field: displayname
Base User Tree: o=mydomain.com,o=employees
User Search Attributes: uid
Group Display Name Field: cn
Base group Tree: =mydomain.com,o=employees
group-Member association uniqueMember
```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder
+-------------------------------+-------------------------------------------------------+
| Configuration                 |                                                       |
+-------------------------------+-------------------------------------------------------+
| hasMemberOfFilterSupport      | 0                                                     |
| hasPagedResultSupport         |                                                       |
| homeFolderNamingRule          |                                                       |
| lastJpegPhotoLookup           | 0                                                     |
| ldapAgentName                 | uid=proxy-user,ou=People,o=mydomain.com,o=employees |
| ldapAgentPassword             | ***                                                   |
| ldapAttributesForGroupSearch  |                                                       |
| ldapAttributesForUserSearch   | uid                                                   |
| ldapBackupHost                |                                                       |
| ldapBackupPort                |                                                       |
| ldapBase                      |                            |o=mydomain.com,o=employees
| ldapBaseGroups                |          o=mydomain.com,o=employees                   |
| ldapBaseUsers                 |                   o=mydomain.com,o=employees          |
| ldapCacheTTL                  | 600                                                   |
| ldapConfigurationActive       | 1                                                     |
| ldapEmailAttribute            |                                                       |
| ldapExperiencedAdmin          | 1                                                     |
| ldapExpertUUIDGroupAttr       |                                                       |
| ldapExpertUUIDUserAttr        |                                                       |
| ldapExpertUsernameAttr        |                                                       |
| ldapGroupDisplayName          | cn                                                    |
| ldapGroupFilter               |                                                       |
| ldapGroupFilterGroups         |                                                       |
| ldapGroupFilterMode           | 0                                                     |
| ldapGroupFilterObjectclass    |                                                       |
| ldapGroupMemberAssocAttr      | uniqueMember                                          |
| ldapHost                  myldapserver.com                                 |
| ldapIgnoreNamingRules         |                                                       |
| ldapLoginFilter               | (uid=%uid)                                            |
| ldapLoginFilterAttributes     |                                                       |
| ldapLoginFilterEmail          | 0                                                     |
| ldapLoginFilterMode           | 0                                                     |
| ldapLoginFilterUsername       | 1                                                     |
| ldapNestedGroups              | 1                                                     |
| ldapOverrideMainServer        | 0                                                     |
| ldapPagingSize                | 500                                                   |
| ldapPort                      | 389                                                   |
| ldapQuotaAttribute            | cn                                                    |
| ldapQuotaDefault              |                                                       |
| ldapTLS                       | 0                                                     |
| ldapUserDisplayName           | displayname                                           |
| ldapUserDisplayName2          |                                                       |
| ldapUserFilter                | (objectclass=person)                                  |
| ldapUserFilterGroups          |                                                       |
| ldapUserFilterMode            | 0                                                     |
| ldapUserFilterObjectclass     | person                                                |
| ldapUuidGroupAttribute        | auto                                                  |
| ldapUuidUserAttribute         | auto                                                  |
| turnOffCertCheck              | 1                                                     |
| useMemberOfToDetectMembership | 1                                                     |
+-------------------------------+-------------------------------------------------------+
+-------------------------------+-------------------------------------------------------+
| Configuration                 | s01                                                   |
+-------------------------------+-------------------------------------------------------+
| hasMemberOfFilterSupport      | 0                                                     |
| hasPagedResultSupport         |                                                       |
| homeFolderNamingRule          |                                                       |
| lastJpegPhotoLookup           | 0                                                     |
| ldapAgentName                 | uid=proxy-user2,ou=People, o=mydomain.com,o=employees
| ldapAgentPassword             | ***                                                   |
| ldapAttributesForGroupSearch  |                                                       |
| ldapAttributesForUserSearch   |                                                       |
| ldapBackupHost                |                                                       |
| ldapBackupPort                |                                                       |
| ldapBase                      |                            |
| ldapBaseGroups                |                                                       |
| ldapBaseUsers                 |                                                       |
| ldapCacheTTL                  | 600                                                   |
| ldapConfigurationActive       | 0                                                     |
| ldapEmailAttribute            |                                                       |
| ldapExperiencedAdmin          | 1                                                     |
| ldapExpertUUIDGroupAttr       |                                                       |
| ldapExpertUUIDUserAttr        |                                                       |
| ldapExpertUsernameAttr        |                                                       |
| ldapGroupDisplayName          | cn                                                    |
| ldapGroupFilter               |                                                       |
| ldapGroupFilterGroups         |                                                       |
| ldapGroupFilterMode           | 0                                                     |
| ldapGroupFilterObjectclass    |                                                       |
| ldapGroupMemberAssocAttr      | uniqueMember                                          |
| ldapHost                      | myldapserver.com                                |
| ldapIgnoreNamingRules         |                                                       |
| ldapLoginFilter               | uid=%uid                                              |
| ldapLoginFilterAttributes     |                                                       |
| ldapLoginFilterEmail          | 0                                                     |
| ldapLoginFilterMode           | 0                                                     |
| ldapLoginFilterUsername       | 1                                                     |
| ldapNestedGroups              | 0                                                     |
| ldapOverrideMainServer        |                                                       |
| ldapPagingSize                | 500                                                   |
| ldapPort                      | 389                                                   |
| ldapQuotaAttribute            |                                                       |
| ldapQuotaDefault              |                                                       |
| ldapTLS                       | 0                                                     |
| ldapUserDisplayName           | displayName                                           |
| ldapUserDisplayName2          |                                                       |
| ldapUserFilter                | (objectclass=*)                                       |
| ldapUserFilterGroups          |                                                       |
| ldapUserFilterMode            | 0                                                     |
| ldapUserFilterObjectclass     |                                                       |
| ldapUuidGroupAttribute        | auto                                                  |
| ldapUuidUserAttribute         | auto                                                  |
| turnOffCertCheck              | 0                                                     |
| useMemberOfToDetectMembership | 1                                                     |
+-------------------------------+-------------------------------------------------------+

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**
Firefox
**Operating system:**
Win 7
### Logs
#### Web server error log
```
[Tue May 17 14:20:42 2016] [error] [client xxx.xxx.xxx.xxx] client denied by server configuration: /var/www/html/owncloud/data/htaccesstest.txt
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
,"time":"2016-05-17T12:21:13+00:00"}
{"reqId":"ehGK1dDAgCEzuc+CVAV6","remoteAddr":"10.26.22.1","app":"PHP","message":"ldap_control_paged_result_response(): No server controls in result at \/var\/www\/html\/owncloud\/apps\/user_ldap\/lib\/ldap.php#70","level":3,"time":"2016-05-17T12:34:36+00:00"}
{"reqId":"ehGK1dDAgCEzuc+CVAV6","remoteAddr":"10.26.22.1","app":"user_ldap","message":"LDAP Login: Could not get user object for DN uid=user,ou=people,o=mydomain.com,o=employees. Maybe the LDAP entry has no set display name attribute?","level":2,"time":"2016-05-17T12:34:36+00:00"}
{"reqId":"ehGK1dDAgCEzuc+CVAV6","remoteAddr":"10.26.22.1","app":"core","message":"Login failed: 'user' (Remote IP: '10.26.22.1')","level":2,"time":"2016-05-17T12:34:36+00:00"}

Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```
To sum up - local users work just fine. I have a problem with ldap users. I have a special user to connect with my ldap

I would appreciate any help. I am fighting with it over 2 weeks now and I've searched around almost whole Internet for an answer.

**Comment by PVince81** [comment]
Weird exception, check this out @blizzz @davicente

**Comment by blizzz** [comment]
> Could not get user object for DN uid=xxx,ou=people,o=domain,o=employees. Maybe the LDAP entry has no set display name attribute?

@pawel1615 did you check this?

**Comment by pawel1615** [comment]
Hi
Thanks for such a quick answer. I have create a new proxy-user in a different tree and set in advance tab:
ldapUserDisplayName : uid
ldapUserDisplayName2: uid 
And it works. It is strange thing: After 2 weeks of stuck on it, just an hour after I've opened a ticket, it works :) 
Anyway. Now ldap users can log in. But in Users menu, instead uid names like "pawel", there are some strange phrases like "616B756C-7A73-612E-7567-757374796E69616B".

**Comment by blizzz** [comment]
>  And it works. It is strange thing: After 2 weeks of stuck on it, just an hour after I've opened a ticket, it works :) 

Glad it is solved for you :) Closing therefore.

> Anyway. Now ldap users can log in. But in Users menu, instead uid names like "pawel", there are some strange phrases like "616B756C-7A73-612E-7567-757374796E69616B". 

Please read https://doc.owncloud.org/server/9.0/admin_manual/configuration_user/user_auth_ldap.html#expert-settings
